File: 3251-maximum-area-of-longest-diagonal-rectangle/maximum-area-of-longest-diagonal-rectangle.cpp
class Solution {
public:
    int areaOfMaxDiagonal(vector<vector<int>>& dimensions) {
        int n = dimensions.size();
        long double maxi = 0;
        int maxiar=0;
        long double  p = 0;
        for (int i = 0; i < n; i++) {
           int  q=dimensions[i][0];
            int r=dimensions[i][1];
            p = (q*q +r*r);           
            p = sqrt(p);

            if (p > maxi || (p==maxi && q*r>maxiar)) {

                maxi = p;
                maxiar=q*r;
            }
        }
        return maxiar;
    }
};

File: 3251-maximum-area-of-longest-diagonal-rectangle/maximum-area-of-longest-diagonal-rectangle.py
import math
class Solution(object):
    def areaOfMaxDiagonal(self, dimensions):
        """
        :type dimensions: List[List[int]]
        :rtype: int
        """
        maxi=0
        maxiarea=0
        n=len(dimensions)
        for i in range(0,n):
            r=dimensions[i][0]
            q=dimensions[i][1]
            p=r*r+q*q
            p=math.sqrt(p)
            if(p > maxi) or (p==maxi and r*q>maxiarea):
                maxi=p
                maxiarea=r*q
        return maxiarea